### ISSUE TITLE
Add an option to build in chroot

### ISSUE THREAD

**Issue by D101101** [archlinuxfr/yaourt#117]
It would be great if yaourt had an option for the build packages in chroot.
For example, I can't build xorg-server-bug865 without chroot.

**Comment by yan12125** [comment]
+1 for this feature.

**Comment by bartoszek** [comment]
Moreover it would greatly improve some split packages, like [python-powerline-git](https://aur.archlinux.org/pkgbase/python-powerline-git/)
Packages which are _self conflicting_ and building them requires install both sets of tools fulfilling all dependencies. But at the end only one variant of the package will be installed and all dependencies for second variant should be removed from the root system.
When building with chroot main system could be capt clean of unnecessary build dependencies.

_think chroot build should be quite easy to implement, find a place to initialise the chroot environment and than switch makepkg to makechrootpkg, will try implement myself when find some free time_

**Comment by yan12125** [comment]
@bartoszek What a coincidence! I've just switched to pacaur as there are too many missing features in yaourt. pacaur supports self-conflicting packages just fine.

Note that you'll need pacaur-git for color support like yaourt.

This table is useful: https://wiki.archlinux.org/index.php/AUR_helpers#Comparison_table

**Comment by vessd** [comment]
@bartoszek Yes, this should be easy. Archlinux already have build scripts : https://wiki.archlinux.org/index.php/DeveloperWiki:Building_in_a_Clean_Chroot

**Comment by bartoszek** [comment]
@yan12125 haven't been using pacaur for some time now - should check how it progresses.
